### DISCUSSION TITLE
double photos after re-import jpg-raw pair

### DISCUSSION THREAD

**Discussion by maja2020** [photoprism/photoprism#4508] [Q&A]
My use case is a camera that produces a jpg-raw pair. Importing pictures works fine. 
When I travel and have to use low bandwith solutions to get my pictures online, I upload/import the JPG only. 

When I import the photo as a pair again, the result  I expected (hoped for) is a single stacked jpg/raw foto pair .

I did the following tests with two different photos:
1. Import the RAW file: it was treated as a new file: I can understand that, since the double photo detection is hash based (sha). Photoprism has no way of recornising this only RAW version is the same foto. By filename, but that's not used for obvious reasons. 

2. Re-Import the JPG-RAW file pair: now the JPG is recognized (hash detection) but the end result is still the same: a JPG and a RAW next to each other in the search UI.

My questions:
- Is this expected behaviour?
- What can I do to correct the situation? Delete originals en re-import the whole batch? 

the import log is attached. Left to photos was test 1, right two photos test 2.

[20240907-photoprism-import-log.txt](https://github.com/user-attachments/files/16919402/20240907-photoprism-import-log.txt)
<img width="1111" alt="2024-09-07-photoprism import result" src="https://github.com/user-attachments/assets/5a4df161-d304-4e1a-b54d-718f9ac7d093">

**Comment by graciousgrey** [comment]
This behavior is to be expected.

Files with exactly the same file name such as IMG_123.jpg and IMG_123.raw will be stacked if they are imported together.

If you import the JPG, it will be moved to the “Originals” folder and given a new file name such as 20240808_124318_16199XXX.jpg. 
If you import the RAW image at a later time, it will not be stacked with the previously imported JPG image. This is to avoid unwanted stacks as many cameras use the same file names and therefore unrelated files can have the same file name.

If you try to import the RAW+JPG pair, the JPG image will be skipped as it is already present in “Originals”. So the result is the same as if you only import the RAW image at a later time.

Currently, you can either [delete the JPG](https://docs.photoprism.app/user-guide/organize/delete/) and then re-import the JPG/RAW pair or, instead of importing, move the RAW file directly to the “Originals” folder in the same subfolder as the associated JPG and rename it so that its name matches the new name of the JPG.

To correct the files that have already been imported, you can rename the files in the Originals folder accordingly and index the folder again.
